src/v/kafka/protocol/types.cc

```diff
@@ -40,4 +40,30 @@ std::ostream& operator<<(std::ostream& os, const uuid& u) {
     return os << u.to_string();
 }
 
+inline std::ostream& operator<<(std::ostream& os, describe_configs_type t) {
+    switch (t) {
+    case describe_configs_type::unknown:
+        return os << "{unknown}";
+    case describe_configs_type::boolean:
+        return os << "{boolean}";
+    case describe_configs_type::string:
+        return os << "{string}";
+    case describe_configs_type::int_type:
+        return os << "{int}";
+    case describe_configs_type::short_type:
+        return os << "{short}";
+    case describe_configs_type::long_type:
+        return os << "{long}";
+    case describe_configs_type::double_type:
+        return os << "{double}";
+    case describe_configs_type::list:
+        return os << "{list}";
+    case describe_configs_type::class_type:
+        return os << "{class}";
+    case describe_configs_type::password:
+        return os << "{password}";
+    }
+    return os << "{unknown}";
+}
```

Comment:
So I get the below compile error with this change.

```
FAILED: src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o 
/usr/bin/ccache /home/nyalia/redpanda/vbuild/llvm/install/bin/clang++ -DBASE64_STATIC_DEFINE -DBOOST_ASIO_HAS_STD_INVOKE_RESULT -DFMT_LOCALE -DSEASTAR_API_LEVEL=6 -DSEASTAR_DEBUG -DSEASTAR_DEBUG_SHARED_PTR -DSEASTAR_DEFAULT_ALLOCATOR -DSEASTAR_SCHEDULING_GROUPS_COUNT=16 -DSEASTAR_SHUFFLE_TASK_QUEUE -DSEASTAR_TYPE_ERASE_MORE -DXXH_PRIVATE_API -DZSTD_STATIC_LINKING_ONLY -I/home/nyalia/redpanda/src/v -I/home/nyalia/redpanda/vbuild/debug/clang/src/v -isystem /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include -isystem /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/hdr -fPIC -march=westmere -fuse-ld=lld -mllvm -inline-threshold=2500 -Wno-unused-lambda-capture -Wno-unused-command-line-argument -stdlib=libc++ -fsanitize=undefined -fsanitize=address -O0 -g -gdwarf-4 -gsplit-dwarf -fPIC -Wall -Wextra -Werror -Wno-missing-field-initializers -U_FORTIFY_SOURCE -DSEASTAR_SSTRING -Werror=unused-result "-Wno-error=#warnings" -fstack-clash-protection -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr -std=c++20 -MD -MT src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o -MF src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o.d -o src/v/kafka/protocol/CMakeFiles/v_kafka_protocol.dir/schemata/protocol.cc.o -c /home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc
In file included from /home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc:2:
In file included from /home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/api_versions_request.h:3:
In file included from /home/nyalia/redpanda/src/v/kafka/protocol/types.h:13:
In file included from /home/nyalia/redpanda/src/v/bytes/bytes.h:14:
In file included from /home/nyalia/redpanda/src/v/bytes/iobuf.h:14:
In file included from /home/nyalia/redpanda/src/v/bytes/details/io_byte_iterator.h:14:
In file included from /home/nyalia/redpanda/src/v/bytes/details/io_fragment.h:14:
In file included from /home/nyalia/redpanda/src/v/bytes/details/out_of_range.h:15:
In file included from /home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/fmt/format.h:48:
/home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/fmt/core.h:1727:3: error: static_assert failed due to requirement 'formattable' "Cannot format an argument. To make type T formattable provide a formatter<T> specialization: https://fmt.dev/latest/api.html#udt"
  static_assert(
  ^
/home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/fmt/core.h:1853:23: note: in instantiation of function template specialization 'fmt::v8::detail::make_arg<true, fmt::basic_format_context<fmt::appender, char>, fmt::v8::detail::type::custom_type, const kafka::describe_configs_type &, 0>' requested here
        data_{detail::make_arg<
                      ^
/home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/fmt/format.h:844:10: note: in instantiation of function template specialization 'fmt::format_arg_store<fmt::basic_format_context<fmt::appender, char>, const seastar::basic_sstring<char, unsigned int, 15, true>, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>>, const bool, const bool, const kafka::describe_configs_source, const bool, const std::vector<kafka::describe_configs_synonym>, const kafka::describe_configs_type, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>>>::format_arg_store<const seastar::basic_sstring<char, unsigned int, 15, true> &, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>> &, const bool &, const bool &, const kafka::describe_configs_source &, const bool &, const std::vector<kafka::describe_configs_synonym> &, const kafka::describe_configs_type &, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>> &>' requested here
  return {args...};
         ^
/home/nyalia/redpanda/vbuild/debug/clang/rp_deps_install/include/fmt/ostream.h:131:15: note: in instantiation of function template specialization 'fmt::make_args_checked<const seastar::basic_sstring<char, unsigned int, 15, true> &, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>> &, const bool &, const bool &, const kafka::describe_configs_source &, const bool &, const std::vector<kafka::describe_configs_synonym> &, const kafka::describe_configs_type &, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>> &, char[125], char>' requested here
         fmt::make_args_checked<Args...>(format_str, args...));
              ^
/home/nyalia/redpanda/vbuild/debug/clang/src/v/kafka/protocol/schemata/protocol.cc:2565:10: note: in instantiation of function template specialization 'fmt::print<char[125], const seastar::basic_sstring<char, unsigned int, 15, true> &, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>> &, const bool &, const bool &, const kafka::describe_configs_source &, const bool &, const std::vector<kafka::describe_configs_synonym> &, const kafka::describe_configs_type &, const std::optional<seastar::basic_sstring<char, unsigned int, 15, true>> &, char>' requested here
    fmt::print(o,
```

Reply:
Looks like you need to declare `std::ostream& operator<<(std::ostream& os, describe_configs_type t);` in types.h

Reply:
(and drop `inline` from the definition)

Reply:
> Looks like you need to declare std::ostream& operator<<(std::ostream& os, describe_configs_type t); in types.h

Wow. Silly mistakes on my part. Thanks!